Tests/SonarScanner.MSBuild.PreProcessor.Tests/Infrastructure/MockSonarQubeServer.cs

```diff
@@ -49,6 +52,16 @@ public void AssertMethodCalled(string methodName, int callCount)
             actualCalls.Should().Be(callCount, "Method was not called the expected number of times");
         }
 
+        public void AssertWarningWritten(string warning)
+        {
+            this.warnings.Should().Contain(warning);
+        }
+
+        public void AssertNoWarningWritten()
+        {
+            this.warnings.Count.Should().Be(0);
```

Comment:
More specific assertion for this special case:
```suggestion
            this.warnings.Should().BeEmpty();
```